### PR TITLE
Remove Cython from `install_requires`.

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Cython is not a dependency of Optuna so I think it should not be listed here even though some optional dependencies don't provide Python 3.9 wheel. 
Refs https://github.com/conda-forge/optuna-feedstock/pull/24

## Description of the changes

Move Cython from `install_requires` to `extra_requires` for tests.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@HideakiImamura Could you review this PR? The document jobs of CI fail time after time, so maybe we should resolve the CI issue, first.

```console
Unexpected failing examples:
/home/docs/project/tutorial/20_recipes/002_multi_objective.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/docs/project/tutorial/20_recipes/002_multi_objective.py", line 123, in <module>
    study.optimize(objective, n_trials=30, timeout=300)
...
urllib.error.HTTPError: HTTP Error 503: Service Unavailable
```

File: setup.py [hunk 1]
```diff
@@ -153,7 +150,13 @@ def get_extras_require() -> Dict[str, List[str]]:
             "botorch>=0.4.0 ; python_version>'3.6'",
             "fastai",
         ],
-        "tests": ["fakeredis", "pytest"],
+        "tests": [
+            # TODO(c-bata): Remove Cython after removing version constraints of sklearn
+            # from 'optional' in extra_requires.
+            "Cython; python_version > '3.8'",
```
Comment (toshihikoyanase) [inline setup.py:156]: According to the comment, this line should be moved to `optional`. The installation of `optional` actually failed due to the following error:

```
ModuleNotFoundError: Please install Cython with a version >= 0.28.5 in order to build a scikit-learn from source.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR.

In my understanding, `Cython` is required by `scikit-learn` if Python version is 3.9. We can see `doctest`, `document`, `example` and `integration` but this PR doesn't add `Cython` to them. Could you update them, please?

File: setup.py (hunk 1 above)
Comment (c-bata) [reply setup.py:156]: Okay. I'm actually not sure we should force all Python 3.9 users to install Cython, but it's acceptable to add it in `optional` dependencies.

**Comment by c-bata** [comment]
Regarding to the `document`, there is no constraints in scikit-learn so that Python 3.9 wheel is available. Thus, I omitted to add Cython in `document`.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Review by HideakiImamura — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**